src/ripple/core/SQLInterface_global.h

```diff
@@ -0,0 +1,204 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CORE_SQLINTERGACE_GLOBAL_H_INCLUDED
+#define RIPPLE_CORE_SQLINTERGACE_GLOBAL_H_INCLUDED
+
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/app/misc/Manifest.h>
+#include <ripple/core/Config.h>
+#include <ripple/core/DatabaseCon.h>
+#include <ripple/overlay/PeerReservationTable.h>
+#include <ripple/peerfinder/impl/Store.h>
+#include <boost/filesystem.hpp>
+
+namespace ripple {
+
+/* Wallet DB */
+
+std::unique_ptr<DatabaseCon>
+makeWalletDB(
+    bool setupFromConfig,
```

Comment:
See previous comment on `setupFromConfig` bool paramter. Consider passing in the `setup` struct directly.